files/scripts/syncd.sh

```diff
@@ -105,6 +105,9 @@ function stopplatform1() {
     fi
 
     if [[ x$sonic_asic_platform != x"mellanox" ]] || [[ x$TYPE != x"cold" ]]; then
+        # Invoke syncd pre shutdown routine
+        /usr/bin/docker exec -i syncd$DEV /usr/bin/syncd_request_pre_shutdown --${TYPE} > /dev/null
```

Comment:
Which PR introduced /usr/bin/syncd_request_pre_shutdown? Calling a non-existing script will cause failure.

Reply:
If the script is not present, the command just throws an error which we redirect to /dev/null.
It should not cause script failure.

Reply:
root@sonic:/home/abc# /usr/bin/docker exec -i syncd0 ls /usr/bin/syncd_request_pre_shutdown
ls: cannot access '/usr/bin/syncd_request_pre_shutdown': No such file or directory           <<<<< **file not present**
root@sonic:/home/abc# 
root@sonic:/home/abc# /usr/bin/docker exec -i syncd0 /usr/bin/syncd_request_pre_shutdown
OCI runtime exec failed: exec failed: container_linux.go:380: starting container process caused: exec: "/usr/bin/syncd_request_pre_shutdown": stat /usr/bin/syncd_request_pre_shutdown: no such file or directory
root@sonic:/home/abc# 

root@sonic:/home/abc# /usr/bin/docker exec -i syncd0 /usr/bin/syncd_request_pre_shutdown --cold > /dev/null
root@sonic:/home/abc#

Reply:
We have the required script added on Cisco platforms, other platforms may add it if required. if not present, it will be a no op.

Reply:
Suggest:

1. call this script from platform plugin directory instead of /usr/bin.
2. Check if the script exists before calling.

Reply:
There is no platform plugin available in this path before syncd_request_shutdown. I can add a plugin as in here if that is acceptable:
https://github.com/sonic-net/sonic-buildimage/pull/13564/commits/2288dbcd075dc54451b7b761d054cfc8ce637093

Reply:
@yxieca , please let me know if the following is fine which does the script check and run only if available:
        PLATFORM=`$SONIC_DB_CLI CONFIG_DB hget 'DEVICE_METADATA|localhost' platform`
        PLATFORM_PRE_SHUTDOWN="/usr/share/sonic/device/$PLATFORM/plugins/syncd_pre_shutdown.sh"
        [ -f $PLATFORM_PRE_SHUTDOWN ] && $PLATFORM_PRE_SHUTDOWN
        
 If you and @abdosi agree, I will update the PR

Reply:
this show work correct ?

`
PLATFORM_PRE_SHUTDOWN="/usr/share/sonic/device/$PLATFORM/plugins/syncd_pre_shutdown.sh"
[ -f $PLATFORM_PRE_SHUTDOWN ] && /usr/bin/docker exec -i syncd$DEV //usr/share/sonic/platform/plugins/syncd_request_pre_shutdown --${TYPE} `

Reply:
yes @abdosi , if @yxieca agrees, I can create the platform plugin like this or call PLATFORM_PRE_SHUTDOWN and let that handle the docker call.

Reply:
ok, please process with above change . it handle @yxieca  request and also we make sure to call from docker context.

Reply:
@abdosi , updated the code, please check.